src/amber.cc

```diff
@@ -25,6 +27,64 @@
 #include "src/vkscript/parser.h"
 
 namespace amber {
+namespace {
+
+// Parse |buffer_id| and return desciptor set and binding using |desc_set|
+// and |binding|. |buffer_id| must be non-empty string. It must be a single
+// non-negative integer or two those integers separated by ':'. For example,
+// ":0", "1", and "2:3" are valid strings for |buffer_id|, but "", "-4",
+// ":-5", ":", "a", and "b:c" are invalid strings for |buffer_id|.
+amber::Result ParseDescriptorSetAndBinding(const std::string& buffer_id,
```

Comment:
This is complicated enough it needs tests. Maybe move it out to a helper and do it there? You could also do something like:
```
Tokenizer t(buffer_id);
auto token = t.nextToken();
if (token.IsInteger()) {
  if (token.AsInt32() < 0)
    return error;

  uint32_t val = token.AsUint32();
  token = t.nextToken();
  if (token.IsEOS() || token.IsEOL()) {
    *desc_set = 0;
    *binding = val;
  }
 *desc_set = val;
}
if (!token.IsString() || token.AsString() != ':')
  return error;

token = t.nextToken();
if (!token.IsInteger() || token.AsInt32() < 0)
  return error;

*binding = token->AsUint32();
```

Reply:
done